diff --git a/dist/cjs/config.cjs b/dist/cjs/config.cjs
index 41c2c24..c93893d 100644
--- a/dist/cjs/config.cjs
+++ b/dist/cjs/config.cjs
@@ -41,6 +41,6 @@ var configSchema = baseConfigSchema.extend({
   routeTreeFileFooter: zod.z.union([
     zod.z.array(zod.z.string()).optional().default([]),
-    zod.z.function().returns(zod.z.array(zod.z.string()))
+    zod.z.custom((value) => typeof value === "function")
   ]).optional(),
   autoCodeSplitting: zod.z.boolean().optional(),
   customScaffolding: zod.z.object({
diff --git a/dist/esm/config.js b/dist/esm/config.js
index f3c745f..883628a 100644
--- a/dist/esm/config.js
+++ b/dist/esm/config.js
@@ -39,6 +39,6 @@ const configSchema = baseConfigSchema.extend({
   routeTreeFileFooter: z.union([
     z.array(z.string()).optional().default([]),
-    z.function().returns(z.array(z.string()))
+    z.custom((value) => typeof value === "function")
   ]).optional(),
   autoCodeSplitting: z.boolean().optional(),
   customScaffolding: z.object({
diff --git a/src/config.ts b/src/config.ts
index 3c5cd52..36471d9 100644
--- a/src/config.ts
+++ b/src/config.ts
@@ -54,7 +54,7 @@ export const configSchema = baseConfigSchema.extend({
   routeTreeFileFooter: z
     .union([
       z.array(z.string()).optional().default([]),
-      z.function().returns(z.array(z.string())),
+      z.custom<() => Array<string>>((value) => typeof value === 'function'),
     ])
     .optional(),
   autoCodeSplitting: z.boolean().optional(),
