diff --git a/lib/unixTerminal.js b/lib/unixTerminal.js
index 368190672e86dbddaa27ad912a51366b41fa0064..3e998950496302461e90834f571c7f74e443dbd6 100644
--- a/lib/unixTerminal.js
+++ b/lib/unixTerminal.js
@@ -28,7 +28,7 @@ var pty = native.module;
 var helperPath = native.dir + '/spawn-helper';
 helperPath = path.resolve(__dirname, helperPath);
-helperPath = helperPath.replace('app.asar', 'app.asar.unpacked');
-helperPath = helperPath.replace('node_modules.asar', 'node_modules.asar.unpacked');
+helperPath = helperPath.replace(/app\.asar(?!\.unpacked)/, 'app.asar.unpacked');
+helperPath = helperPath.replace(/node_modules\.asar(?!\.unpacked)/, 'node_modules.asar.unpacked');
 var DEFAULT_FILE = 'sh';
 var DEFAULT_NAME = 'xterm';
 var DESTROY_SOCKET_TIMEOUT_MS = 200;
